Add note about root.render microtask scheduling (#7717)
reactjs/react.dev: src/content/reference/react-dom/client/createRoot.md

* If you call `render` on the same root more than once, React will update the DOM as necessary to reflect the latest JSX you passed. React will decide which parts of the DOM can be reused and which need to be recreated by ["matching it up"](/learn/preserving-and-resetting-state) with the previously rendered tree. Calling `render` on the same root again is similar to calling the [`set` function](/reference/react/useState#setstate) on the root component: React avoids unnecessary DOM updates.

> * Although rendering is synchronous once it starts, calling `root.render(...)` schedules the render in a microtask. This means code after `root.render()` will run before any effects (`useLayoutEffect`, `useEffect`) are fired. This is usually fine and rarely needs adjustment. In rare cases where effect timing matters, you can wrap `root.render(...)` in [`flushSync`](https://react.dev/reference/react-dom/client/flushSync) to ensure the initial render runs fully synchronously.

Reviewer: ```suggestion
* Although rendering is synchronous once it starts, `root.render(...)` is not. This means code after `root.render()` may run before any effects (`useLayoutEffect`, `useEffect`) of that specific render are fired. This is usually fine and rarely needs adjustment. In rare cases where effect timing matters, you can wrap `root.render(...)` in [`flushSync`](https://react.dev/reference/react-dom/client/flushSync) to ensure the initial render runs fully synchronously.
```

The exact timing of when `root.render()` starts is either an implementation detail or decided by the context in which it runs e.g. if it's called in `flushSync` it's not in the next microtask.

A small code example would help here e.g.

```
root.render(...)
// Will not include the HTML from this specific `root.render` call
console.log(document.body.innerHTML)
```

Author: Good points, I agree, thanks. I will reflect it 👍
